### PR TITLE
Homework 9 Vahan Harutyunyan Annotation Using custom exception and generics

### PR DESCRIPTION
**Pull request by Vahann**

### CODE REVIEW

File: src/annotation/myannotation/IntervalAnnotationProcessor.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+                        e.printStackTrace();
+                    }
+
+                }
+
+
+            }
+        }
+    }
+
```
Comment (khemma11) [inline src/annotation/myannotation/IntervalAnnotationProcessor.java:53]: Comments must be deleted before opening PR

**Review by khemma11 — COMMENTED** [review]
Good job

File: src/annotation/myannotation/IntervalAnnotationProcessor.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+//    private static void initField(Object object) {
```
Comment (khemma11) [inline src/annotation/myannotation/IntervalAnnotationProcessor.java:54]: Comments must be deleted before opening PR

File: src/annotation/myannotation/AdulthoodAnnotationProcessor.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+import annotation.exception.ExpectedTypeException;
+
+import java.lang.reflect.Field;
+import java.time.LocalDate;
+import java.time.Period;
+
+public class AdulthoodAnnotationProcessor {
+
+    public static <T> void definitionFieldsForAnnotationLength(T object) throws IllegalAccessException{
```
Comment (LevonAloyan) [inline src/annotation/myannotation/AdulthoodAnnotationProcessor.java:11]: Not a meaningful name for this method.

File: src/annotation/myannotation/AdulthoodAnnotationProcessor.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
+package annotation.myannotation;
+
+import annotation.exception.ExpectedTypeException;
+
+import java.lang.reflect.Field;
+import java.time.LocalDate;
+import java.time.Period;
+
+public class AdulthoodAnnotationProcessor {
```
Comment (LevonAloyan) [inline src/annotation/myannotation/AdulthoodAnnotationProcessor.java:9]: Please create an AnnotationProcessor interface with the process method then implement all your processors from that one

File: src/annotation/myannotation/AdulthoodAnnotationProcessor.java (hunk 3 above)
Comment (LevonAloyan) [inline src/annotation/myannotation/AdulthoodAnnotationProcessor.java:11]: Here you don't need to use static methods, to use polymorphism

File: src/annotation/myannotation/AdulthoodAnnotationProcessor.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+                if (temp instanceof LocalDate){
+                    String value = (String) fieldName.get(object).toString();
+
+//                System.out.println("Field value: " + value);
+                    LocalDate localDate = LocalDate.parse(value);
+                    Period period = Period.between(localDate, LocalDate.now());
+//                System.out.println(period.getYears());
+
+                    if (period.getYears() < 18) {
+                        System.err.println("В классе: " + aClass + "\n в поле: "
```
Comment (LevonAloyan) [inline src/annotation/myannotation/AdulthoodAnnotationProcessor.java:31]: Please create an array of error messages and add your error message to it.

File: src/annotation/myannotation/EmailAnnotationProcessor.java [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+//                System.out.println("Field value: " + value);
+                    if (!EmailValidator.validate(value)) {
+                        System.err.println("В классе: " + aClass + "\n в поле: "
+                                + field + " значение: " + value + "\n не соответствует требованиям аннотации: "
+                                + Email.class.getName() + "\n");
+                    }
+                }else {
+                    try {
+                        throw  new ExpectedTypeException();
```
Comment (LevonAloyan) [inline src/annotation/myannotation/EmailAnnotationProcessor.java:34]: why you are throwing exceptions here?

**Pull request closed without merging**